src/runtime_src/core/tools/common/ReportMemory.cpp

```diff
@@ -389,6 +422,33 @@ ReportMemory::writeReport( const xrt_core::device* /*_pDevice*/,
     }
   }
 
+  bool sk_mem_is_present = _pt.get_child("mem_topology.board.memory.sk_memories",empty_ptree).size() > 0 ? true:false;
+  if (sk_mem_is_present) {
+    _output << std::endl;
+    _output << "  Soft Kernel Memory Status" << std::endl;
+    _output << boost::format("  %-20s%-16s%-16s%-16s\n") % "  HostBO Count" % "MapBO Count" % "UnmapBO Count" % "FreeBO Count";
+
+    try {
+      std::string hbo_count, mapbo_count, unmapbo_count, freebo_count;
+      for (auto& v : _pt.get_child("mem_topology.board.memory.sk_memories",empty_ptree)) {

```

Comment:
This for look seems overly complicated and possibly error prone. 

Is there a reason why you didn't consider the following:

```
auto & sk_memories_pt = _pt.get_child("mem_topology.board.memory.sk_memories",empty_ptree);
const std:string &hbo_count = sk_memories_pt.get<std::string>("sk_hbo","");
const std:string &mapbo_count = sk_memories_pt.get<std::string>("sk_mapbo","");
const std:string &unmapbo_count = sk_memories_pt.get<std::string>("sk_unmapbo","");
const std:string &freebo_count = sk_memories_pt.get<std::string>(sk_freebo,"");

```

Reply:
This suggestions are really helpful. Fixed. Thanks.